test/expectations/expectations.go

```diff
@@ -83,6 +84,18 @@ func ExpectApplied(ctx context.Context, c client.Client, objects ...client.Objec
 	}
 }
 
+func ExpectStatusConditions(ctx context.Context, c client.Client, obj status.Object, conditions ...status.Condition) {
```

Comment:
Now that I look at this, why does this even take a context and client? Did you mean to call ExpectGet?